File: typescript/src/messages/verdict_request_for_url.ts
import { JsonProperty, JsonObject } from "typescript-json-serializer";
import { Kind, Message } from "./message";

@JsonObject()
export class VerdictRequestForUrl extends Message {
  public constructor(url: URL, guid: string, session_id: string, use_cache?: boolean, use_hash_lookup?: boolean) {
    super(Kind.VerdictRequestForUrl);
    this.url = url.toString();
    this.session_id = session_id;
    this.guid = guid;
    this.use_cache = use_cache;
    this.use_hash_lookup = use_hash_lookup;
  }

  @JsonProperty() public url: string;
  @JsonProperty() public guid: string;
  @JsonProperty() public session_id: string;
  @JsonProperty() public use_hash_lookup?: boolean;
  @JsonProperty() public use_cache?: boolean;  
}

File: ruby/lib/vaas/verdict_request_for_url.rb
module VAAS
    class VerdictRequestForUrl
  
      attr_reader :url, :session_id, :guid
  
      def initialize(url, session_id, guid)
        @kind = "VerdictRequestForUrl"
        @url = url
        @session_id = session_id,
        @guid = guid
      end
  
    end
  end
  